spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DeploymentPropertiesResolver.java

```diff
@@ -584,50 +584,53 @@ Affinity getAffinityRules(Map<String, String> kubernetesDeployerProperties) {
 	}
 
 	Container getInitContainer(Map<String, String> kubernetesDeployerProperties) {
-		Container container = null;
 		KubernetesDeployerProperties deployerProperties = bindProperties(kubernetesDeployerProperties,
 				this.propertyPrefix + ".initContainer", "initContainer");
 
-		if (deployerProperties.getInitContainer() == null) {
-			String containerName = PropertyParserUtils.getDeploymentPropertyValue(kubernetesDeployerProperties,
-					this.propertyPrefix + ".initContainer.containerName");
-
-			String imageName = PropertyParserUtils.getDeploymentPropertyValue(kubernetesDeployerProperties,
-					this.propertyPrefix + ".initContainer.imageName");
-
-			String commands = PropertyParserUtils.getDeploymentPropertyValue(kubernetesDeployerProperties,
```

Comment:
**[NOTE]** This was a bug lurking in the shadows.. It was not parsing the csv command string and just passing into the single string builder method. It is fixed now to do so.